lib/devicetrust/testenv/fake_device_service.go

```diff
@@ -458,17 +504,46 @@ func (s *FakeDeviceService) AuthenticateDevice(stream devicepb.DeviceTrustServic
 		return trace.Wrap(err)
 	}
 
+	newCerts := &devicepb.UserCertificates{}
+	if !hasWebToken {
+		newCerts.X509Der = []byte("<insert augmented X.509 cert here")
+		newCerts.SshAuthorizedKey = []byte("<insert augmented SSH cert here")
+	}
+
 	err = stream.Send(&devicepb.AuthenticateDeviceResponse{
 		Payload: &devicepb.AuthenticateDeviceResponse_UserCertificates{
-			UserCertificates: &devicepb.UserCertificates{
-				X509Der:          []byte("<insert augmented X.509 cert here"),
-				SshAuthorizedKey: []byte("<insert augmented SSH cert here"),
-			},
+			UserCertificates: newCerts,
 		},
 	})
 	return trace.Wrap(err)
 }
 
+func (s *FakeDeviceService) spendDeviceWebToken(webToken *devicepb.DeviceWebToken, dev *storedDevice) error {
+	const invalidWebTokenMessage = "invalid device web token"
+
+	for i, token := range s.deviceWebTokens {
+		if token.id != webToken.Id {
+			continue
+		}
+
+		// Spend token regardless of outcome.
+		s.deviceWebTokens = slices.Delete(s.deviceWebTokens, i, i+1)
+
+		// Validate token info.
+		switch {
+		case token.token != webToken.Token:
+			return trace.AccessDenied(invalidWebTokenMessage)
+		case token.expectedDeviceID != dev.pb.Id:
+			return trace.AccessDenied(invalidWebTokenMessage)
+		}
```

Comment:
Again, plain ifs might be easier to understand

Reply:
Ack  👍